modules/core/Scene.h

```diff
@@ -35,12 +35,12 @@ namespace rt {
 		DllExport CScene(const CScene&) = delete;
 		DllExport ~CScene(void) = default;
 		DllExport const CScene& operator=(const CScene&) = delete;
-	  
+
 		// ------------ TO IMPLEMENT ------------
 		/// @todo implement
-		DllExport void save(const std::string& fileName) const {}
+		DllExport bool save(const std::string& fileName, const Mat& image) const;
 		/// @todo implement
-		DllExport void load(const std::string& fileName) {}
+		DllExport Mat load(const std::string& fileName) const;
```

Comment:
Yes I agree with your point here.
In this case, we would need to implement a serialization mechanism and store these in a custom file, some sort of project file would make sense (i.e.: `myTestScene.rt`). This is a bit of a more complicated task because we would have to implement serialization to all subclasses of the scene (every geometry, every light, every shader, etc...). I think there's a library that helps with implementing serialization really well (boost) but this would mean an extra dependency which we would have to understand in further detail. If I get the green light, I can research this a bit and maybe discuss it in our future meeting.
For now, I'll revert the changes here and move the implementation somewhere else.

Reply:
Yes, this serialization idea would be serious task for the future, that may take a lot of efforts. 
I am against incorporating the Boost library, instead maybe it would make sense to develop own file format...
The scene is just a collection of 3 types of objects: Geometry, Lights and Cameras. Geometry contain also shaders, which may be considered as a fourth distinct type of objects (because 2 different objects may have the same shader). Saving lights and cameras are straightforward - this classes may have its own serialization - deserialization methods, which take the file descriptor as an argument. The main challenge is to serialize geometry with corresponding shaders.